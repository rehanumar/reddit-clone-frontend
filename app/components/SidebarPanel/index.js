/**
 *
 * SidebarPanel
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Card from 'components/Card';

const StyledHead = styled.div`
  font-size: 10px;
  font-weight: 700;
  letter-spacing: 0.5px;
  line-height: 12px;
  background-color: ${props => props.theme.sidebarWidgetHeaderColor};
  border-radius: 3px 3px 0 0;
  color: ${props => props.theme.sidebarWidgetTitleColor};
  display: flex;
  fill: ${props => props.theme.sidebarWidgetTitleColor};
  padding: 0 12px 12px;
`;

const StyledHeading = styled.h2`
  padding: 12px 0 0;
  font-size: 14px;
  line-height: 18px;
  font-weight: 700;
`;

const StyledBody = styled.div`
  padding: 12px;
  vertical-align: baseline;
`;

function SidebarPanel({ heading, children }) {
  return (
    <Card style={{ width: '310px', border: '1px solid #ccc' }}>
      <StyledHead>
        <StyledHeading>{heading}</StyledHeading>
      </StyledHead>
      <StyledBody>{children}</StyledBody>
    </Card>
  );
}

SidebarPanel.propTypes = {
  heading: PropTypes.string.isRequired,
  children: PropTypes.elementType.isRequired,
};

export default SidebarPanel;
