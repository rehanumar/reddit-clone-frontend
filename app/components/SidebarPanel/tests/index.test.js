/**
 *
 * Tests for SidebarPanel
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import SidebarPanel from '../index';

describe('<SidebarPanel />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<SidebarPanel />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Expect to have additional unit tests specified', () => {
    const props = {
      heading: 'About Community',
    };
    const {
      container: { firstChild },
    } = render(<SidebarPanel {...props} />);
    expect(firstChild).toMatchSnapshot();
  });

  /**
   * Unskip this test to use it
   *
   * @see {@link https://jestjs.io/docs/en/api#testskipname-fn}
   */
  it.skip('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<SidebarPanel />);
    expect(firstChild).toMatchSnapshot();
  });
});
