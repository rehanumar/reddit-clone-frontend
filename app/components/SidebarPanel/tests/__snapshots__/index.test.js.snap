// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidebarPanel /> Expect to have additional unit tests specified 1`] = `
<div
  class="Card-neputd-0 ivdfyp"
  style="width: 370px;"
>
  <div
    class="SidebarPanel__StyledHead-sc-17k1qi6-0 ixVMhF"
  >
    <h2
      class="SidebarPanel__StyledHeading-sc-17k1qi6-1 ljJdBj"
    >
      About Community
    </h2>
  </div>
  <div
    class="SidebarPanel__StyledBody-sc-17k1qi6-2 gsiLXt"
  />
</div>
`;
